CMakeLists.txt

```diff
@@ -16,11 +22,66 @@ if(ALPAKA_ACC_GPU_CUDA_ENABLE)
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};--expt-extended-lambda")
 endif()
 
-# Use doxygen for api doc when available.
-find_package(Doxygen QUIET)
-if(DOXYGEN_FOUND)
+
+add_library(${PROJECT_NAME} INTERFACE)
+target_include_directories(${PROJECT_NAME}
+  INTERFACE
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+target_link_libraries(${PROJECT_NAME}
+  INTERFACE
+  alpaka::alpaka)
+add_library(vikunja::vikunja ALIAS ${PROJECT_NAME})
+
+
+if(${CMAKE_PROJECT_NAME} STREQUAL ${PROJECT_NAME})
+  #########################################
+  ######### Install Configuration #########
+  #########################################
+
+  include(CMakePackageConfigHelpers)
+  write_basic_package_version_file(
+    "${PROJECT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
+    VERSION 0.1
+    COMPATIBILITY AnyNewerVersion
+    )
+
+  include(GNUInstallDirs)
+  install(TARGETS ${PROJECT_NAME} alpaka
```

Comment:
Why are we installing alpaka here? What if I already have an installed alpaka on my system?

Reply:
Because alpaka is a header-only library, it does not install any file. This needs to by another command (at least in the solution, which I found). The command is necessary to export the target definition. Otherwise the following code does not work:

```
target_link_libraries(${PROJECT_NAME}
  INTERFACE
  alpaka::alpaka)
  ```  
If you have a better solution, please tell me it.

Reply:
`alpaka::alpaka` should be already available if `find_package(alpaka)` was successful. Or am I missing something?

Reply:
Could be :thinking: I added `find_library("alpaka")` in the config file afterwards. This could solve the problem. I test it.

Reply:
The command does not install alpaka. It's necessary for the target configuration. I tested it with a alpaka, which is installed in my environment.

If I remove the `alpaka`, I get the following error: `CMake Error: install(EXPORT "vikunjaTargets" ...) includes target "vikunja" which requires target "alpaka" that is not in any export set.`

Reply:
Interesting, I'll look into this. I think this may be a bug in alpaka's CMake scripts.

Reply:
Do you want to test it, before merging or should I keep it. It looks ugly, but I didn't saw any malicious side effect.

Reply:
You can keep it for now, I don't know when I'll find the time. If I can solve this in alpaka the fix is very easily ported to vikunja.

Reply:
I looked into this a bit further and it needs to be fixed in alpaka itself. Right now alpaka's CMake installation script doesn't export the alpaka target. This allows us to reconfigure alpaka after installing it but comes with the issue we see here. I'll open an issue on the alpaka tracker to discuss this.

Reply:
I introduced a new target for internal linkage and removed the export of alpaka. Now, the linkage to alpaka is done, if the package was found in a external project. See commit  75eaa7edc6f8ff77169f1678d8b409a7db0b3cd1